model/atmosphere/dycore/src/icon4py/model/atmosphere/dycore/nh_solve/solve_nonhydro.py

```diff
@@ -724,80 +715,50 @@ def run_predictor_step(
             )
 
         p_dthalf = 0.5 * dtime
+        cell_domain = h_grid.domain(dims.CellDim)
+        edge_domain = h_grid.domain(dims.EdgeDim)
+        vertex_domain = h_grid.domain(dims.VertexDim)
+        end_cell_end = self.grid.end_index(cell_domain(h_grid.Zone.END))
 
-        end_cell_end = self.grid.get_end_index(
-            dims.CellDim, h_grid.HorizontalMarkerIndex.end(dims.CellDim)
-        )
+        second_halo_cell = cell_domain(h_grid.Zone.HALO_LEVEL_2)
+        start_cell_halo_level_2 = self.grid.start_index(second_halo_cell)
+        end_cell_haloe_level_2 = self.grid.end_index(second_halo_cell)
 
-        start_cell_local_minus2 = self.grid.get_start_index(
-            dims.CellDim, h_grid.HorizontalMarkerIndex.local(dims.CellDim) - 2
-        )
-        end_cell_local_minus2 = self.grid.get_end_index(
-            dims.CellDim, h_grid.HorizontalMarkerIndex.local(dims.CellDim) - 2
+        start_vertex_lb_level_2 = self.grid.start_index(
+            vertex_domain(h_grid.Zone.LATERAL_BOUNDARY_LEVEL_2)
         )
+        end_vertex_halo = self.grid.end_index(vertex_domain(h_grid.Zone.HALO))
 
-        start_vertex_lb_plus1 = self.grid.get_start_index(
-            dims.VertexDim, h_grid.HorizontalMarkerIndex.lateral_boundary(dims.VertexDim) + 1
-        )  # TODO: check
-        end_vertex_local_minus1 = self.grid.get_end_index(
-            dims.VertexDim, h_grid.HorizontalMarkerIndex.local(dims.VertexDim) - 1
-        )
+        start_cell_lb = self.grid.start_index(cell_domain(h_grid.Zone.LATERAL_BOUNDARY))
+        end_cell_lb_level_4 = self.grid.end_index(cell_domain(h_grid.Zone.LATERAL_BOUNDARY_LEVEL_4))
 
-        start_cell_lb = self.grid.get_start_index(
-            dims.CellDim, h_grid.HorizontalMarkerIndex.lateral_boundary(dims.CellDim)
-        )
-        end_cell_nudging_minus1 = self.grid.get_end_index(
-            dims.CellDim,
-            h_grid.HorizontalMarkerIndex.nudging(dims.CellDim) - 1,
+        start_edge_lb_level_7 = self.grid.start_index(
```

Comment:
More consistency renaming

```suggestion
        start_edge_lateral_boundary_level_7 = self.grid.start_index(
```